lte/gateway/python/magma/subscriberdb/crypto/EC.py

```diff
@@ -0,0 +1,158 @@
+            format=serialization.PrivateFormat.Raw,
+            encryption_algorithm=serialization.NoEncryption(),
+        )
+
+    def generate_sharedkey(self, ext_pubkey):
+        """Generates the shared key
+        """
+        extpubkey = X25519PublicKey.from_public_bytes(ext_pubkey)
+        return self.PrivKey.exchange(extpubkey)
+
+
+class ECDH_SECP256R1(object):
+    """wrapper around Python cryptography library to handle an ECDH  exchange over
+    a NIST secp256r1 elliptic curve
+    private key is handled within a DER-encoded PKCS8 structure
+    public key is handled as a compressed point bytes buffer according to ANSI X9.62
+    """
+
+    def __init__(self, loc_privkey=None, _raw_keypair=None):
+        """Initializes the class object
+        """
+        if loc_privkey:
+            self.PrivKey = serialization.load_der_private_key(
+                loc_privkey,
+                password=None,
+                backend=_backend,
+            )
+            if not hasattr(self.PrivKey, 'curve') or not isinstance(self.PrivKey.curve, ec.SECP256R1):
+                raise(CMException('invalid secp256r1 private key'))
+        elif isinstance(_raw_keypair, (tuple, list)) and len(_raw_keypair) == 2:
+            self._load_raw_keypair(*_raw_keypair)
+        else:
+            self.generate_keypair()
+
+    def _load_raw_keypair(self, privkey, pubkey):
+        """Initializes the priv key
+        """
+        self.PrivKey = ec.EllipticCurvePrivateNumbers(
+                       private_value=int_from_bytes(privkey),
+                       public_numbers=ec.EllipticCurvePublicKey.from_encoded_point(
+                       curve=ec.SECP256R1(),
+                       data=pubkey,
+                       ).public_numbers(),
+        ).private_key(backend=_backend)
+
+    def generate_keypair(self):
+        """Generates public/private keypair
+        """
+        self.PrivKey = ec.generate_private_key(
+                       curve=ec.SECP256R1(),
+                       backend=_backend,
+        )
+
+    def get_pubkey(self):
+        """Generates public key
+        """
+        return self.PrivKey.public_key().public_bytes(
+                  format=serialization.PublicFormat.CompressedPoint,
+                  encoding=serialization.Encoding.X962,
+        )
+
+    def get_privkey(self):
+        """Generates private key
+        """
+        return self.PrivKey.private_bytes(
+                  encoding=serialization.Encoding.DER,
+                  format=serialization.PrivateFormat.PKCS8,
+                  encryption_algorithm=serialization.NoEncryption(),
+        )
+
+    def generate_sharedkey(self, ext_pubkey):
+        """Generates shared key
+        """
+        extpubkey = ec.EllipticCurvePublicKey.from_encoded_point(
+                       curve=ec.SECP256R1(),
+                       data=ext_pubkey,
+        )
+        return self.PrivKey.exchange(ec.ECDH(), extpubkey)
+
+
+def KDF(sharedinfo, sharedkey):
```

Comment:
public non-member functions should be at the top of the file

Reply:
Fixed the same